src/pages/login/LoginForm/LoginForm.test.tsx

```diff
@@ -0,0 +1,193 @@
+import { render, screen, fireEvent, act, waitFor } from '@testing-library/react'
+import { AxiosResponse } from 'axios'
+import { createMemoryHistory } from 'history'
+import React from 'react'
+import { Provider } from 'react-redux'
+import { Router } from 'react-router-dom'
+
+import { AuthService } from '../../../API/AuthService'
+import { store } from '../../../store/store'
+import { ROUTES } from '../../../utils/constants'
+
+import { LoginForm } from './index'
+
+jest.mock('../../../API/AuthService')
+
+describe('given LoginForm component', () => {
+  it('should render button with disabled state if both username and password is not filled', async () => {
+    render(
+      <Provider store={store}>
+        <LoginForm />
+      </Provider>,
+    )
+    expect(screen.getByRole('button', { name: /Log in/i })).toBeDisabled()
+  })
+
+  it('should render button with disabled state if username is not filled ', async () => {
+    render(
+      <Provider store={store}>
+        <LoginForm />
+      </Provider>,
+    )
+
+    const passwordField = screen.queryByPlaceholderText('Password')
+
+    await act(async () => {
+      fireEvent.change(passwordField as HTMLElement, { target: { value: 'password' } })
+    })
```

Comment:
If you expect the value to be in the document, you should use `get`.
You can use `query` like `expect(screen.queryByText()).not.toBeInTheDocument()`

```suggestion
    const passwordField = screen.getByPlaceholderText('Password')

    await act(async () => {
      fireEvent.change(passwordField, { target: { value: 'password' } })
    })
```